DashAI/back/core/schema_fields/int_field.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional
+
+from pydantic import Field
+from typing_extensions import Annotated
+
+
+def int_field(
+    description: str,
+    default: int,
+    minimum: Optional[int] = None,
+    maximum: Optional[int] = None,
+):
+    """Function to create a pydantic-like integer type.
+
+    Parameters
+    ----------
+    description: str
+        Description of the field.
+    default: int
+        The default value to show to the user.
+    minimum: Optional int
+        Minimum value of the field.
+    maximum: Optional int
+        Maximum value of the field.
+
+    Returns
+    -------
+    type[int]
+        A pydantic-like type to represent the integer.
+
+    Raises
+    ------
+    ValidationError
+        If the value of the field is less than the minimum.
+    ValidationError
+        If the value of the field is greater than the maximum.
+    """
+    params = {"description": description, "default": default}
+    if minimum:
+        params["ge"] = minimum
+    if maximum:
+        params["le"] = maximum
+    return Annotated[
+        int,
+        Field(**params),
+    ]
```

Comment:
Here mypy is claiming to me that the types do not match. How about passing it the arguments directly (which by type checking, should always match or be None (which as I saw in the field docstring, is a possible value for the arguments))?

```suggestion
    return Annotated[
        int,
        Field(
            description=description,
            default=default,
            ge=ge,
            gt=gt,
            le=le,
            lt=lt,
        ),
    ]
```

Reply:
It's a better option, i use the dictionary because i wasn't sure that the Field function could handle the Nones.